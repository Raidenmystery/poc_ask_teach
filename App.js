import React from "react";
import styled from "styled-components/native";

const Container = styled.View`
  flex: 1;
  align-items: center;
  justify-content: center;
  background-color: #fff;
`;

const Title = styled.Text`
  font-size: 24px;
  font-weight: bold;
  color: #fc00c4;
`;

const Button = styled.TouchableOpacity`
  padding: 10px 20px;
  background-color: #8f8fbb;
  border-radius: 4px;
`;

const ButtonText = styled.Text`
  font-size: 16px;
  color: #fff;
`;

const App = () => {
  return (
    <Container>
      <Title>Hello, world!</Title>
      <Button onPress={() => alert("Button pressed")}>
        <ButtonText>Press me!</ButtonText>
      </Button>
    </Container>
  );
};

export default App;
