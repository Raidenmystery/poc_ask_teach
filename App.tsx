import { STORYBOOK_ENABLED } from "@env";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";

import StorybookUIRoot from "./.storybook/Storybook";
import FormPage from "./pages/FormPage";
import ListPage from "./pages/ListPage";
import QRCameraPage from "./pages/QRCameraPage";

export default function App() {
  const Stack = createNativeStackNavigator();

  if (STORYBOOK_ENABLED === "true") return StorybookUIRoot();

  return (
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="List"
        screenOptions={{ headerShown: false }}
      >
        <Stack.Screen name="Form" component={FormPage} />
        <Stack.Screen name="List" component={ListPage} />
        <Stack.Screen name="QRCamera" component={QRCameraPage} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
