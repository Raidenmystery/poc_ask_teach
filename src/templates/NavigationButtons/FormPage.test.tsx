import React from "react";
import renderer from "react-test-renderer";

import App from "./index";

describe("<App />", () => {
  it("has 5 child", () => {
    const tree = renderer.create(<App navigation={{}} type="List" />).toJSON();
    expect(tree?.children?.length).toBe(2);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<App navigation={{}} type="List" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
