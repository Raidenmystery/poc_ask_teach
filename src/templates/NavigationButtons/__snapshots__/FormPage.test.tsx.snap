// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "gap": 15,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#047a27",
        "borderRadius": 10,
        "padding": 12,
        "width": 150,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Go to Form
    </Text>
  </View>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#047a27",
        "borderRadius": 10,
        "padding": 12,
        "width": 150,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Go to QRCamera
    </Text>
  </View>
</View>
`;
